### ISSUE TITLE
SSL Client hanging in `wait_until_socket_is_ready`

### ISSUE THREAD

**Issue by aardvarkk** [yhirose/cpp-httplib#263]
I'm using macOS Catalina and trying to use the SSL Client in v0.2.5 to make a POST request. The call seems to hang inside of `wait_until_socket_is_ready` regardless of if I set `CPPHTTPLIB_USE_POLL` or not. When `select` fails `errno` appears to be set to 0x24, which I believe maps to `EINPROGRESS`. I'm not sure what exactly is going on but I'm quite confused as to what I'm doing wrong.

**Comment by yhirose** [comment]
@aardvarkk, thank you for the report. I am actually using Catalina on my iMac. I run unit tests with both OpenSSL 1.0.2t and 1.1.1d installed from brew, and confirmed that all the unit tests passed successfully.
I would like to know more detailed about the problem on your machine. When you have time, could you run the unit tests by simpley `cd test; make` and see what will happen?
Thanks a lot!

**Comment by aardvarkk** [comment]
Yes, I should be able to run those tests tonight! I will report the results back here. Thanks for your attention!

**Comment by aardvarkk** [comment]
OK -- I just ran the tests and all of them seemed to pass. 

```
[----------] Global test environment tear-down
[==========] 95 tests from 22 test cases ran. (36241 ms total)
[  PASSED  ] 95 tests.
```

I'm going to try simplifying my code to get a small reproducible example if I can.

**Comment by aardvarkk** [comment]
I got it figured out! The problem was I was accidentally putting some of the path into the hostname. I'm working with Mailgun, and their URLs take the following form:

`https://api.mailgun.net/v3/sandboxabc123.mailgun.org`

I wasn't reading carefully enough and didn't notice the early slashes in the URL -- I just saw the trailing `.mailgun.org` and thought that was the HOST without paying enough attention. After correcting the host to be only `api.mailgun.net` and taking the rest of the URL and prepending it to the path the calls are working correctly. 

Thanks for your help and attention!
